pkg/build/builder/util.go

```diff
@@ -26,3 +30,67 @@ func getBuildEnvVars(build *api.Build) map[string]string {
 	}
 	return envVars
 }
+
+type BuildConfigClientInterface interface {
+	GetBuildConfig(namespace, name string) (*buildapi.BuildConfig, error)
+	UpdateBuildConfig(buildConfig *buildapi.BuildConfig) error
+}
+
+type BuildClientInterface interface {
+	CreateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error)
+	CreateBuildFromConfig(config *buildapi.BuildConfig, revision *buildapi.SourceRevision, imageSubstitutions map[string]string) error
+	CreateBuildWithImmutableId(config *buildapi.BuildConfig) error
+	UpdateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error)
+}
+
+// OSClientBuildClient is a BuildClientInterface which delegates to the OpenShift client interfaces
+type OSClientBuildClient struct {
+	Client osclient.Interface
+}
+
+// CreateBuild creates build using OpenShift client.
+func (c OSClientBuildClient) CreateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error) {
+	return c.Client.Builds(namespace).Create(build)
+}
+
+// UpdateBuild updates build using the OpenShift client.
+func (c OSClientBuildClient) UpdateBuild(namespace string, build *buildapi.Build) (*buildapi.Build, error) {
+	return c.Client.Builds(namespace).Update(build)
+}
+
+// CreateBuild creates a build using the OpenShift client.
+func (c OSClientBuildClient) CreateBuildFromConfig(config *buildapi.BuildConfig, revision *buildapi.SourceRevision, imageSubstitutions map[string]string) error {
+	build := buildutil.GenerateBuildFromConfig(config, revision)
+	for originalImage, newImage := range imageSubstitutions {
+		buildutil.SubstituteImageReferences(build, originalImage, newImage)
+	}
+	if _, err := c.Client.Builds(config.Namespace).Create(build); err != nil {
+		glog.V(2).Infof("Error creating build for buildConfig %v: %v", config.Name, err)
+		return err
+	}
+	return nil
+}
+
+// TODO will implement this in a future pull, want to get the refactor delivered before
+// too much rebasing happens.
+// CreateBuildWithImmutableId creates a build using the OpenShift client and the current immutable imageid
+// from any ImageRepository that is associated to the BuildConfig by an ImageChangeTrigger
+func (c OSClientBuildClient) CreateBuildWithImmutableId(config *buildapi.BuildConfig) error {
```

Comment:
I don't like that name, but unfortunately Go does not allow function overloading :disappointed:. But if we'll stick with it please change `Id` -> `ID` to be consistent with everywhere else.